lib/lib.d.ts

```diff
@@ -349,55 +349,29 @@ interface String {
 
     /**
       * Matches a string with a regular expression, and returns an array containing the results of that search.
-      * @param regexp A variable name or string literal containing the regular expression pattern and flags.
+      * @param regexp A string containing the regular expression pattern or a regular expression object.
       */
-    match(regexp: string): RegExpMatchArray | null;
+    match(regexp: RegExp | string): RegExpMatchArray | null;
 
     /**
-      * Matches a string with a regular expression, and returns an array containing the results of that search.
-      * @param regexp A regular expression object that contains the regular expression pattern and applicable flags.
-      */
-    match(regexp: RegExp): RegExpMatchArray | null;
-
-    /**
-      * Replaces text in a string, using a regular expression or search string.
-      * @param searchValue A string that represents the regular expression.
+      * Replaces text in a string, using a regular expression or search string. Only regular expression with global flag replaces all occurrences.
```

Comment:
Can you reword this to "When given a regular expression with its global flag set, this method replaces all occurrences. Otherwise, only the first match is replaced."